client/src/controllers/UpgradeController.test.js

```diff
@@ -2,9 +2,6 @@ import { Application } from '@hotwired/stimulus';
 
 import { UpgradeController } from './UpgradeController';
 
-// https://stackoverflow.com/a/51045733
-const flushPromises = () => new Promise(setImmediate);
```

Comment:
The update to this SO answer suggests now using `process.nextTick`. What led you to choose `requestAnimationFrame` in most cases you updated in this PR?

Reply:
nextTick is a Node API and makes sense for testing Node, but it won't work for JSDom only tests in some cases. 

Assuming we want a more 'correct' browser test, animation frame seemed more appropriate. It's also what Stimulus uses in their own test suite, although they don't use Jest.

Reply:
Makes sense, thanks!